src/main/java/frc/robot/Autonomous.java

```diff
@@ -0,0 +1,28 @@
+package frc.robot;
+
+import frc.robot.subsystems.drivetrain.Drivetrain;
+import frc.robot.subsystems.ball.Ball;
+
+public class Autonomous {
+    
+    Drivetrain drivetrain;
+    Ball ball;
+
+    static final double WHEEL_DIAMETER = 18.84955592;
+    
+    Autonomous(Drivetrain driveSubsystem, Ball ballSubsystem) {
+        this.drivetrain = driveSubsystem;
+        this.ball = ballSubsystem;
+    }
+
+    public void hitNRun() throws InterruptedException {
+        this.ball.shoot();
+        Thread.sleep(5000);
+        this.ball.cancelShooting();
+        this.drivetrain.setSetpoint(-120 * (WHEEL_DIAMETER * 128)); // Distance in inches * Wheel diameter in inches * ticks per rotation
+        while(!this.drivetrain.atSetpoint()){
+            this.drivetrain.drivePID();
```

Comment:
Please add a `Thread.sleep(20)` (or some other short period) in here, or at the very least a `Thread.yield()`.

Reply:
After setting the setpoint? Setting the setpoint doesn't actually do anything (afaik), it just sets the setpoint of the pid controller. What drivePID is drives the robot, calculating the speed based on the feedback from the encoders and the setpoint. The pid controller was changed this year, so instead of using a PIDOutput, instead there is a method called calculate that calculates the output of the pid controller, and then you pass that to the (in this case) tankDrive method which actually drives the robot.

Reply:
Oh wait you're talking about the loop. lol